### PR TITLE
Update SSHCluster usage in benchmarks with new CUDAWorker

### PR DESCRIPTION
**Pull request by pentschev**
Updates usage of `SSHCluster` according to changes in https://github.com/dask/distributed/pull/5191.

### CODE REVIEW

**Comment by pentschev** [comment]
This PR is still waiting on https://github.com/dask/distributed/pull/3907 getting merged.

**Comment by pentschev** [comment]
Still waiting on https://github.com/dask/distributed/pull/3907 .

**Comment by pentschev** [comment]
I updated the PR and the `name` condition here @charlesbluca , could you take a look as well?

File: dask_cuda/benchmarks/utils.py [hunk 1]
```diff
@@ -206,30 +206,21 @@ def get_cluster_options(args):
...
-            worker_options["device_memory_limit"] = args.device_memory_limit
-        if args.ucx_net_devices:
-            worker_options["ucx_net_devices"] = args.ucx_net_devices
-
         cluster_kwargs = {
             "connect_options": {"known_hosts": None},
-            "scheduler_options": {"protocol": args.protocol},
-            "worker_module": "dask_cuda.dask_cuda_worker",
-            "worker_options": worker_options,
+            "scheduler_options": {"protocol": args.protocol, "port": 8786},
```
Comment (charlesbluca) [inline dask_cuda/benchmarks/utils.py:211]: Just so I understand better - why do we want to explicitly set the scheduler port to 8786?

File: dask_cuda/benchmarks/utils.py [hunk 2]
```diff
@@ -206,30 +206,21 @@ def get_cluster_options(args):
...
+            "worker_class": "dask_cuda.CUDAWorker",
+            "worker_options": {
+                "protocol": args.protocol,
```
Comment (charlesbluca) [inline dask_cuda/benchmarks/utils.py:214]: Is `protocol` being handled by Distributed when creating the workers? Don't see anything in `CUDAWorker` that suggests we need to pass this argument to the constructor

**Review by charlesbluca — COMMENTED** [review]
Looks good overall, thanks for adding in the `name` fix 😄 just a couple questions:

File: dask_cuda/benchmarks/utils.py (hunk 2 above)
Comment (pentschev) [reply dask_cuda/benchmarks/utils.py:214]: Yes, it's passed via `CUDAWorker`'s `kwargs`.

File: dask_cuda/benchmarks/utils.py (hunk 1 above)
Comment (charlesbluca) [reply dask_cuda/benchmarks/utils.py:211]: That makes sense - thanks for the clarification!

**Review by charlesbluca — APPROVED** [review]

**Comment by pentschev** [comment]
This is now ready for review. I've verified it works:

<details><summary>Sample result</summary>

```
$ python dask_cuda/benchmarks/local_cupy.py -d 0,1,2,3,4,5,6,7 --multi-node --hosts dgx13,dgx13,dgx14
distributed.deploy.ssh - INFO - distributed.http.proxy - INFO - To route to workers diagnostics web server please install jupyter-server-proxy: python -m pip install jupyter-server-proxy
distributed.deploy.ssh - INFO - distributed.scheduler - INFO - Clear task state
distributed.deploy.ssh - INFO - distributed.scheduler - INFO -   Scheduler at:  tcp://10.33.227.163:8786
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.163:34611'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.163:46231'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.163:40839'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.163:39261'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.163:44699'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.163:43081'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.163:41553'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.163:42265'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.164:36231'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.164:38385'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.164:34751'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.164:46277'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.164:43871'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.164:39925'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.164:45527'
distributed.deploy.ssh - INFO - distributed.nanny - INFO -         Start Nanny at: 'tcp://10.33.227.164:39429'
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Import preload module: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Run preload setup click command: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.worker - INFO -       Start worker at:  tcp://10.33.227.163:42593
distributed.deploy.ssh - INFO - distributed.preloading - INFO - Run preload setup click command: dask_cuda.initialize
distributed.deploy.ssh - INFO - distributed.worker - INFO -       Start worker at:  tcp://10.33.227.164:33719
Roundtrip benchmark
--------------------------
Operation          | transpose_sum
User size          | 10000
User second size   | 1000
User chunk-size    | 2500
Compute shape      | (10000, 10000)
Compute chunk-size | (2500, 2500)
Ignore-size        | 1.00 MiB
Protocol           | tcp
Device(s)          | 0,1,2,3,4,5,6,7
Worker Thread(s)   | 1
==========================
Wall-clock         | npartitions
--------------------------
741.70 ms          | 16
401.54 ms          | 16
474.88 ms          | 16
==========================
(w1,w2)            | 25% 50% 75% (total nbytes)
--------------------------
(0-0,0-4)            | 503.84 MiB/s 503.84 MiB/s 503.84 MiB/s (47.68 MiB)
(0-0,1-2)            | 307.30 MiB/s 307.30 MiB/s 307.30 MiB/s (47.68 MiB)
(0-0,1-5)            | 352.49 MiB/s 352.49 MiB/s 352.49 MiB/s (47.68 MiB)
(0-1,0-4)            | 412.72 MiB/s 450.57 MiB/s 488.43 MiB/s (95.37 MiB)
(0-2,0-3)            | 477.32 MiB/s 477.32 MiB/s 477.32 MiB/s (47.68 MiB)
(0-2,0-7)            | 229.05 MiB/s 229.05 MiB/s 229.05 MiB/s (47.68 MiB)
(0-2,1-6)            | 201.33 MiB/s 201.33 MiB/s 201.33 MiB/s (47.68 MiB)
(0-3,0-2)            | 440.86 MiB/s 440.86 MiB/s 440.86 MiB/s (47.68 MiB)
(0-3,1-4)            | 240.14 MiB/s 240.14 MiB/s 240.14 MiB/s (47.68 MiB)
(0-4,0-0)            | 547.29 MiB/s 547.29 MiB/s 547.29 MiB/s (47.68 MiB)
(0-4,0-1)            | 332.11 MiB/s 412.50 MiB/s 492.89 MiB/s (95.37 MiB)
(0-5,1-0)            | 258.74 MiB/s 258.74 MiB/s 258.74 MiB/s (47.68 MiB)
(0-5,1-2)            | 209.05 MiB/s 209.05 MiB/s 209.05 MiB/s (47.68 MiB)
(0-6,1-4)            | 216.67 MiB/s 216.67 MiB/s 216.67 MiB/s (47.68 MiB)
(0-7,0-2)            | 235.83 MiB/s 235.83 MiB/s 235.83 MiB/s (47.68 MiB)
(0-7,1-2)            | 203.69 MiB/s 203.69 MiB/s 203.69 MiB/s (47.68 MiB)
(0-7,1-4)            | 300.24 MiB/s 300.24 MiB/s 300.24 MiB/s (47.68 MiB)
(1-0,0-5)            | 233.64 MiB/s 233.64 MiB/s 233.64 MiB/s (47.68 MiB)
(1-0,1-3)            | 448.04 MiB/s 448.04 MiB/s 448.04 MiB/s (47.68 MiB)
(1-0,1-6)            | 454.55 MiB/s 454.55 MiB/s 454.55 MiB/s (47.68 MiB)
(1-1,1-7)            | 327.01 MiB/s 327.01 MiB/s 327.01 MiB/s (47.68 MiB)
(1-2,0-0)            | 248.36 MiB/s 248.36 MiB/s 248.36 MiB/s (47.68 MiB)
(1-2,0-5)            | 240.94 MiB/s 240.94 MiB/s 240.94 MiB/s (47.68 MiB)
(1-2,0-7)            | 221.89 MiB/s 221.89 MiB/s 221.89 MiB/s (47.68 MiB)
(1-3,1-0)            | 484.33 MiB/s 484.33 MiB/s 484.33 MiB/s (47.68 MiB)
(1-3,1-5)            | 322.91 MiB/s 322.91 MiB/s 322.91 MiB/s (47.68 MiB)
(1-4,0-3)            | 227.95 MiB/s 227.95 MiB/s 227.95 MiB/s (47.68 MiB)
(1-4,0-6)            | 202.38 MiB/s 202.38 MiB/s 202.38 MiB/s (47.68 MiB)
(1-4,0-7)            | 341.02 MiB/s 341.02 MiB/s 341.02 MiB/s (47.68 MiB)
(1-5,0-0)            | 165.27 MiB/s 165.27 MiB/s 165.27 MiB/s (47.68 MiB)
(1-5,1-3)            | 307.15 MiB/s 307.15 MiB/s 307.15 MiB/s (47.68 MiB)
(1-6,0-2)            | 222.80 MiB/s 222.80 MiB/s 222.80 MiB/s (47.68 MiB)
(1-6,1-0)            | 490.94 MiB/s 490.94 MiB/s 490.94 MiB/s (47.68 MiB)
(1-7,1-1)            | 306.49 MiB/s 306.49 MiB/s 306.49 MiB/s (47.68 MiB)
```

</details>

However, after the above finishes, the cluster doesn't exit cleanly and dies 30 seconds later:

<details><summary>Close timeout traceback</summary>

```python-traceback
tornado.application - ERROR - Exception in callback <bound method Client._heartbeat of <Client: 'tcp://10.33.227.163:8786' processes=16 threads=16, memory=1.97 TiB>>
Traceback (most recent call last):
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/tornado/ioloop.py", line 905, in _run
    return self.callback()
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/client.py", line 1172, in _heartbeat
    self.scheduler_comm.send({"op": "heartbeat-client"})
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/batched.py", line 136, in send
    raise CommClosedError(f"Comm {self.comm!r} already closed.")
distributed.comm.core.CommClosedError: Comm <TCP (closed) Client->Scheduler local=tcp://127.0.0.1:57628 remote=tcp://dgx13:8786> already closed.
tornado.application - ERROR - Exception in callback functools.partial(<bound method IOLoop._discard_future_result of <tornado.platform.asyncio.AsyncIOMainLoop object at 0x7f8e2597d6d0>>, <Task finished name='Task-126' coro=<SpecCluster._correct_state_internal() done, defined at /datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/deploy/spec.py:325> exception=OSError('Timed out trying to connect to tcp://10.33.227.163:8786 after 30 s')>)
ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/comm/core.py", line 284, in connect
    comm = await asyncio.wait_for(
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/asyncio/tasks.py", line 494, in wait_for
    return fut.result()
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/comm/tcp.py", line 410, in connect
    convert_stream_closed_error(self, e)
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/comm/tcp.py", line 126, in convert_stream_closed_error
    raise CommClosedError(f"in {obj}: {exc.__class__.__name__}: {exc}") from exc
distributed.comm.core.CommClosedError: in <distributed.comm.tcp.TCPConnector object at 0x7f8dea2a65b0>: ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/tornado/ioloop.py", line 741, in _run_callback
    ret = callback()
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/tornado/ioloop.py", line 765, in _discard_future_result
    future.result()
  File "dask_cuda/benchmarks/local_cupy.py", line 315, in run
    await client.shutdown()
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/deploy/cluster.py", line 439, in __aexit__
    await f
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/deploy/spec.py", line 417, in _close
    await self._correct_state()
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/deploy/spec.py", line 332, in _correct_state_internal
    await self.scheduler_comm.retire_workers(workers=list(to_close))
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/core.py", line 785, in send_recv_from_rpc
    comm = await self.live_comm()
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/core.py", line 742, in live_comm
    comm = await connect(
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/comm/core.py", line 308, in connect
    raise OSError(
OSError: Timed out trying to connect to tcp://10.33.227.163:8786 after 30 s
ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/comm/core.py", line 284, in connect
    comm = await asyncio.wait_for(
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/asyncio/tasks.py", line 494, in wait_for
    return fut.result()
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/comm/tcp.py", line 410, in connect
    convert_stream_closed_error(self, e)
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/comm/tcp.py", line 126, in convert_stream_closed_error
    raise CommClosedError(f"in {obj}: {exc.__class__.__name__}: {exc}") from exc
distributed.comm.core.CommClosedError: in <distributed.comm.tcp.TCPConnector object at 0x7f8dea2a65b0>: ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "dask_cuda/benchmarks/local_cupy.py", line 380, in <module>
    main()
  File "dask_cuda/benchmarks/local_cupy.py", line 376, in main
    asyncio.get_event_loop().run_until_complete(run(args))
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/tornado/ioloop.py", line 741, in _run_callback
    ret = callback()
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/tornado/ioloop.py", line 765, in _discard_future_result
    future.result()
  File "dask_cuda/benchmarks/local_cupy.py", line 315, in run
    await client.shutdown()
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/deploy/cluster.py", line 439, in __aexit__
    await f
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/deploy/spec.py", line 417, in _close
    await self._correct_state()
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/deploy/spec.py", line 332, in _correct_state_internal
    await self.scheduler_comm.retire_workers(workers=list(to_close))
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/core.py", line 785, in send_recv_from_rpc
    comm = await self.live_comm()
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/core.py", line 742, in live_comm
    comm = await connect(
  File "/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/comm/core.py", line 308, in connect
    raise OSError(
OSError: Timed out trying to connect to tcp://10.33.227.163:8786 after 30 s
/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/deploy/spec.py:663: RuntimeWarning: coroutine 'wait_for' was never awaited
  cluster.close(timeout=10)
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
/datasets/pentschev/miniconda3/envs/rn-112-21.12.210930/lib/python3.8/site-packages/distributed/deploy/spec.py:663: RuntimeWarning: coroutine 'SpecCluster._close' was never awaited
  cluster.close(timeout=10)
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
```

</details>

Given this works despite the unclean exit, the low-priority of this task and the prior knowledge of the rabbit hole that Distributed closing/finalizers is, I won't go through that now. If anyone feels like doing that, have fun!

**Review by charlesbluca — APPROVED** [review]
Yeah, that is strange behavior, considering there are certainly other areas of the benchmarks that have clean up issues and this doesn't impact the actual times, I'm in agreement to merge this and potentially dig down that rabbit hole later on with a follow up PR in Distributed.

Thanks @pentschev 🙂

**Pull request merged**